paimon-core/src/main/java/org/apache/paimon/append/AppendOnlyTableCompactionCoordinator.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.append;
+
+import org.apache.paimon.CoreOptions;
+import org.apache.paimon.annotation.VisibleForTesting;
+import org.apache.paimon.data.BinaryRow;
+import org.apache.paimon.io.DataFileMeta;
+import org.apache.paimon.table.AppendOnlyFileStoreTable;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/** {@link AppendOnlyFileStoreTable} compact coordinator. */
+public class AppendOnlyTableCompactionCoordinator {
+    private final long targetFileSize;
+    private final int minFileNum;
+    private final int maxFileNum;
+
+    private final Map<BinaryRow, PartitionCompactCoordinator> partitionCompactCoordinators =
+            new HashMap<>();
+
+    public AppendOnlyTableCompactionCoordinator(AppendOnlyFileStoreTable table) {
+        CoreOptions coreOptions = table.coreOptions();
+        this.targetFileSize = coreOptions.targetFileSize();
+        this.minFileNum = coreOptions.compactionMinFileNum();
+        this.maxFileNum = coreOptions.compactionMaxFileNum();
+    }
+
+    public void notifyNewFiles(BinaryRow partition, List<DataFileMeta> files) {
+        partitionCompactCoordinators
+                .computeIfAbsent(partition, PartitionCompactCoordinator::new)
+                .addFiles(
+                        files.stream()
+                                .filter(file -> file.fileSize() < targetFileSize)
+                                .collect(Collectors.toList()));
+    }
+
+    // generate compaction task to the next stage
+    public List<AppendOnlyCompactionTask> compactPlan() {
+        return partitionCompactCoordinators.values().stream()
```

Comment:
judge:
- `PartitionCompactCoordinator ` is empty, remove it.
- If `PartitionCompactCoordinator` is a file, plan 10 times, no other files, just remove it. 
- If `PartitionCompactCoordinator` has some file, plan 5 times, no other files, just compact them.